torch/_inductor/cudagraph_trees.py

```diff
@@ -754,10 +754,12 @@ def __init__(
         self.device = device_index
         self.stack_traces = stack_traces
         self.stream = stream
-        # If we are inlining builtin nn modules we will re-record if static inputs change
-        # if not we should error because dynamo should have recompiled in this case
+
+        # Enable re-record a cudagraph when static tensor address changed.
+        # if not we should error when it changed.
         self.rerecord_if_static_inputs_change = (
             torch._dynamo.config.inline_inbuilt_nn_modules
+            or torch._inductor.config.cudagraph_support_input_mutation
```

Comment:
Does this work without enabling inline_inbuilt_nn_modules? I don't think it will because AOT autograd needs to mark the correct argument indices as static.

Can you point me to the examples you guys are running into? I had a more narrow solution that involves marking only certain params as static in dynamo.